src/sinks/elasticsearch/common.rs

```diff
@@ -143,15 +144,20 @@ impl ElasticsearchCommon {
             metric_config.metric_tag_values,
         );
 
+        let service_type = config.opensearch_service_type;
+
         let version = if let Some(version) = *version {
             version
+        } else if service_type == OpenSearchServiceType::Serverless {
+            // Amazon OpenSearch Serverless does not support the cluster-version API
+            8
```

Comment:
I think this is worth a warning and a mention in the config docs since users might find the connection between service type and api version unexpected.